sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/MessageListenerStrategy.java

```diff
@@ -105,4 +108,17 @@ public void commitOnClose(KafkaConsumer<K, V> consumer) {
   public Map<String, String> forcedConfigToApply() {
     return Collections.emptyMap();
   }
+
+  protected void startTraceContext(ConsumerRecord<K, V> consumerRecord) {
+    Header header = consumerRecord.headers().lastHeader(ConsumerTracing.TOKEN_HEADER);
+    if (header != null) {
+      TraceContext.storeTraceToken(new String(header.value()));
+    } else {
+      TraceContext.createNewTraceToken();
+    }
+  }
```

Comment:
Afaik, we should not keep the same context but make a connection between the received trace token and a new one for this new synchronous process.